Code/GraphMol/AddHs.cpp

```diff
@@ -562,7 +561,7 @@ void removeHs(RWMol &mol, bool implicitOnly, bool updateExplicitCount,
     Atom *atom = mol.getAtomWithIdx(currIdx);
     idxMap[origIdx] = currIdx;
     ++origIdx;
-    if (atom->getAtomicNum() == 1) {
+    if (atom->getAtomicNum() == 1 && atom->getDegree()) {
```

Comment:
I think this is a time for a warning.  This is mildly unexpected outcome that removeHs doesn't ;)

I.e. WARNING: not removing zero degree hydrogen and index 10

This is now really removeConnectedHs, correct?

Reply:
What happens with [H][H] btw, does only one get removed?

Reply:
good point about the warning.
I'll add a test case for `[H][H]`

Reply:
I should be more careful when answering things from my phone... :-)
The `[H][H]` case is already documented: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/MolOps.h#L209
and tested: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/molopstest.cpp#L613